### PR TITLE
Es6 lib config

### PR DESCRIPTION
**Pull request by gtrufitt**
Reverts the revert and fixes the polyfill loading in Karma.

### CODE REVIEW

**Review by SiAdcock — APPROVED** [review]
:godmode:

**Comment by sndrs** [comment]
this looks v useful - can we merge it?

**Comment by gustavpursche** [comment]
@sndrs Unfortunately not. It was on `master` already, but we had to revert, because so far there isn't a bulletproof way of injecting the polyfills into Karma.

**Comment by gtrufitt** [comment]
@gustavpursche Had a look with Alex this morning - it's possible that this *is* the way to inject the polyfills but there is a problem with Promises... @sndrs was going to take a look :)

**Comment by gustavpursche** [comment]
@gtrufitt Great. Looking forward to see this show-stopper in production :)

**Comment by gtrufitt** [comment]
image

OH MY GOODNESS HALLELUJAH

File: static/src/javascripts/lib/config.spec.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+// We need to hoist setting the window.guardian.config to code under test above import
+// (but can't do it with the jest mocks that get auto-hoisted)
+// eslint-disable-next-line import/first
+import config from './config';
+
+jest.mock('lib/pad', () => jest.fn(arg => arg));
+
+afterAll(() => {
+    window.guardian.config = prevConfig;
```
Comment (SiAdcock) [inline static/src/javascripts/lib/config.spec.js]: I don't think this is strictly necessary as each test spec file should run in its own environment. I tested this by removing the `afterAll` here and `console.log`ging in all other test files to ensure the `page` property was not present. 

Also I think `prevConfig` at this point is pointing to `window.guardian.config` anyway, so this is not doing anything.

Comment (sndrs) [inline static/src/javascripts/lib/config.spec.js]: cool

**Review by SiAdcock — APPROVED** [review]
Re-approved 👍 :1st_place_medal:

**Pull request merged**